src/main/java/uk/gov/companieshouse/web/accounts/controller/cic/CicApprovalController.java

```diff
@@ -0,0 +1,99 @@
+package uk.gov.companieshouse.web.accounts.controller.cic;
+
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import uk.gov.companieshouse.web.accounts.annotation.NextController;
+import uk.gov.companieshouse.web.accounts.annotation.PreviousController;
+import uk.gov.companieshouse.web.accounts.controller.BaseController;
+import uk.gov.companieshouse.web.accounts.controller.smallfull.StepsToCompleteController;
+import uk.gov.companieshouse.web.accounts.exception.ServiceException;
+import uk.gov.companieshouse.web.accounts.model.cic.CicApproval;
+import uk.gov.companieshouse.web.accounts.service.cic.CicApprovalService;
+import uk.gov.companieshouse.web.accounts.service.transaction.TransactionService;
+import uk.gov.companieshouse.web.accounts.validation.ValidationError;
+
+@Controller
+@PreviousController(CicReviewController.class)
+@NextController(StepsToCompleteController.class)
+@RequestMapping("/company/{companyNumber}/transaction/{transactionId}/company-accounts/{companyAccountsId}/cic/approval")
+public class CicApprovalController extends BaseController {
+
+
+    private static final String APPROVAL = "cicApproval";
+    private static final String TRANSACTION_ID = "transaction_id";
+    private static final String COMPANY_ACCOUNTS_ID = "company_accounts_id";
+
+    @Autowired
+    private TransactionService transactionService;
+
+    @Autowired
+    private CicApprovalService cicApprovalService;
+
+    @Override
+    protected String getTemplateName() {
+        return "cic/cicApproval";
+    }
+
+    @GetMapping
+    public String getApproval(@PathVariable String companyNumber,
+        @PathVariable String transactionId,
+        @PathVariable String companyAccountsId,
+        Model model) {
+
+        addBackPageAttributeToModel(model, companyNumber, transactionId, companyAccountsId);
+
+        model.addAttribute(APPROVAL, new CicApproval());
```

Comment:
We should try to fetch an existing approval here because a user may have already provided one, rather than providing a new instance each time.